core/domain/learner_playlist_services_test.py

```diff
@@ -179,6 +180,46 @@ def test_mark_exploration_to_be_played_later(self):
             self._get_all_learner_playlist_exp_ids(
                 self.user_id), exp_ids)
 
+        # Empty the learner playlist again.
+        for exp_id in exp_ids:
+            learner_playlist_services.remove_exploration_from_learner_playlist(
+                self.user_id, exp_id)
+        self.assertEqual(
+            self._get_all_learner_playlist_exp_ids(
+                self.user_id), [])
+
+        # Subscribe to exploration.
+        subscription_services.subscribe_to_exploration(
+            self.user_id, 'SUBSCRIBED_ID')
+
+        # Now if we try to add the same exploration, it shouldn't be added.
```

Comment:
I am a little confused about the conceptual reasoning behind this. Why wouldn't a subscribed exploration be added to the playlist?

Reply:
I was thinking that line 102 of the learner_playlist_services requires the exploration id to not be subscribed to. So, this is what I test: In 191-193, I subscribe to an exploration. Then in 195-197, I try to add it to the playlist, but since line 102 evaluates to false, it shouldn't be added. Is this right?

Reply:
Ah, I see, interesting -- I think you're right! Thanks for that. It looks like "subscribed" means "created/edited by" and I mistakenly thought it meant "is followed by" instead.

Reply:
Yep!